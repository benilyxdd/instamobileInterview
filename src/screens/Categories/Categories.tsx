import React from 'react';
import { View, FlatList } from 'react-native';

import { categories, recipes } from '../../../constants/mockData';
import Card from '../../components/Card';

interface Props {
	categories: typeof categories;
	recipes: typeof recipes;
}

const Categories: React.FunctionComponent<Props> = ({
	categories,
	recipes,
}) => {
	return (
		<View>
			<FlatList
				testID="flatlist"
				data={categories}
				keyExtractor={item => item.id.toString()}
				renderItem={({ item }) => {
					const itemCount = recipes.filter(
						recipes => recipes.categoryId === item.id,
					).length;

					return (
						<Card
							size="md"
							title={item.name}
							imageUrl={item.photo_url}
							subtitle={`${itemCount} recipes`}
						/>
					);
				}}
			/>
		</View>
	);
};

export default Categories;
