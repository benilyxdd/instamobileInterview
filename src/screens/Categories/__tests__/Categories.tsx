import React from 'react';
import { create } from 'react-test-renderer';

import Categories from '../Categories';

const screen = create(<Categories />);

describe('Categories Screen', () => {
	describe('Components', () => {
		it('snapshot', () => {
			expect(screen).toMatchSnapshot();
		});

		it('renders a flatlist', () => {
			const flatlist = screen.root.findByProps({
				testID: 'flatlist',
			}).props;
		});
	});
});
