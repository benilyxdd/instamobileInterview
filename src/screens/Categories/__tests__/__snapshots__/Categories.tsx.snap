// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Screen Components snapshot 1`] = `
<View>
  <Text>
    Categories Screen
  </Text>
</View>
`;
